src/v/cloud_storage_clients/xml_sax_parser.cc

```diff
@@ -385,8 +385,22 @@ seastar::future<client::list_bucket_result> parse_from_stream(
                    [&stream] { return stream.eof(); },
                    [&stream, &p] {
                        return stream.read().then(
-                         [&p](ss::temporary_buffer<char>&& chunk) {
-                             p.parse_chunk(std::move(chunk));
+                         [&p](ss::temporary_buffer<char> chunk) {
+                             /*
+                              * Seastar may return chunks of 128KB which is very
+                              * close to our max allocation limit. But when this
+                              * is handed off to to libxml the data may be
+                              * incorporated into a buffer that needs resizing
+                              * and exceed the max allocation limit. So break
+                              * this up into smaller chunks for processing.
+                              */
+                             constexpr auto max_part_size = 16_KiB;
+                             while (!chunk.empty()) {
+                                 auto part_size = std::min(
+                                   max_part_size, chunk.size());
+                                 p.parse_chunk(chunk.share(0, part_size));
```

Comment:
Nit: would probably have been nicer inside the parse_chunk before feeding it to the C api. No need to reallocate a temporary_buffer structure.

Reply:
@nvartolomei totally, i wish I'd thought of that!